classes/user_enrolment_repository.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace local_augmented_teacher;
+
+class user_enrolment_repository
+{
+    /**
+     * Get user enrolments by course ids
+     * @param array $course_ids
+     * @return array
+     * @throws \dml_exception
+     */
+    public function get_by_course_ids(array $course_ids): array {
+        global $DB;
+
+        $sql = 'SELECT id, status, userid
+                             FROM {user_enrolments}
+                             WHERE enrolid IN (SELECT id
+                                               FROM {enrol}
+                                               WHERE courseid IN ('.implode(', ', $course_ids).'))';
```

Comment:
Be aware of SQL injections.

Please replace with $this->db->get_in_or_equal($course_ids).